### PR TITLE
dts: nordic: nrf54h20_enga: add new peripherals

### PR DESCRIPTION
**Pull request by gmarull**
Add new peripherals to the nRF54H20 devicetree. Also add necessary changes to the hal_nordic module glue code or SoC Kconfig files.

### CODE REVIEW

**Review by hubertmis — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by 57300 — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@nordicjm can I merge this?

**Comment by nordicjm** [comment]
> @nordicjm can I merge this?

yes

**Comment by gmarull** [comment]
> > @nordicjm can I merge this?
> 
> yes

note it slightly touches boards

File: dts/common/nordic/nrf54h20_enga.dtsi [hunk 1]
```diff
@@ -243,28 +396,344 @@
...
+				cc-num = <6>;
+				interrupts = <419 NRF_DEFAULT_IRQ_PRIORITY>;
+				max-bit-width = <32>;
+				prescaler = <0>;
+			};
+
+			i2c130: i2c@9a5000 {
+				compatible = "nordic,nrf-twim";
+				reg = <0x9a5000 0x1000>;
+				status = "disabled";
```
Comment (anangl) [inline dts/common/nordic/nrf54h20_enga.dtsi:465]: All I2C nodes are lacking `easydma-maxcnt-bits = <15>;`.

File: dts/common/nordic/nrf54h20_enga.dtsi [hunk 2]
```diff
@@ -243,28 +396,344 @@
...
+
 			uart136: uart@9d5000 {
 				compatible = "nordic,nrf-uarte";
 				reg = <0x9d5000 0x1000>;
 				status = "disabled";
-				current-speed = <115200>;
 				interrupts = <469 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
+
+			timer136: timer@9d2000 {
```
Comment (anangl) [inline dts/common/nordic/nrf54h20_enga.dtsi:691]: `timer136` and `timer137` should be located before `i2c136` to keep the address order.

File: dts/common/nordic/nrf54h20_enga.dtsi (hunk 1 above)
Comment (gmarull) [reply dts/common/nordic/nrf54h20_enga.dtsi:465]: would you mind sending a patch?

File: dts/common/nordic/nrf54h20_enga.dtsi (hunk 2 above)
Comment (gmarull) [reply dts/common/nordic/nrf54h20_enga.dtsi:691]: ditto

File: dts/common/nordic/nrf54h20_enga.dtsi (hunk 1 above)
Comment (gmarull) [reply dts/common/nordic/nrf54h20_enga.dtsi:465]: wait, I see this hasn't been merged, I'll do it myself

**Review by 57300 — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

File: dts/common/nordic/nrf54h20_enga.dtsi [hunk 3]
```diff
@@ -243,28 +396,351 @@
...
+
 			uart136: uart@9d5000 {
 				compatible = "nordic,nrf-uarte";
 				reg = <0x9d5000 0x1000>;
 				status = "disabled";
-				current-speed = <115200>;
 				interrupts = <469 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
+
+			i2c137: i2c@9d6000 {
```
Comment (anangl) [inline dts/common/nordic/nrf54h20_enga.dtsi:718]: Add `easydma-maxcnt-bits = <15>;` also here.

**Review by anangl — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Pull request merged**